website_cms/models/cms_page.py

```diff
@@ -0,0 +1,660 @@
+            "website_cms.cms_manager" group bypass this;
+        `nav` to show nav-included items;
+        `types_ids` to limit listing to specific page types;
+        `types_ref` to limit listing to specific page types;
+            by xmlid refs
+        `order` to override ordering by sequence;
+        `path` to search in a specific path instead of
+            just listing current item's children;
+        `incl_tags` to include items related by same tags;
+        `tag_ids` to filter upone specific tags.
+
+        By default filter w/ `list_types_ids` if valued.
+        """
+        item = item or self
+        base_domain = []
+        # use specific `published` items but bypass it if manager
+        if published is not None and \
+                not self.env.user.has_group('website_cms.cms_manager'):
+            base_domain.append(('website_published', '=', published))
+
+        if nav is not None:
+            base_domain.append(('nav_include', '=', nav))
+
+        if types_ref:
+            if isinstance(types_ref, basestring):
+                types_ref = (types_ref, )
+            types_ids = [self.env.ref(x).id for x in types_ref]
+
+        types_ids = types_ids or (
+            item.list_types_ids and item.list_types_ids.ids)
+
+        if types_ids:
+            base_domain.append(
+                ('type_id', 'in', types_ids)
+            )
+
+        hierarchy_domain = []
+        if path is None:
+            hierarchy_domain.append(('parent_id', '=', item.id))
+        else:
+            hierarchy_domain.append(('path', '=like', path + '%'))
+
+        # handle domain for tags
+        tags_domain = []
+        if incl_tags or tag_ids:
+            # we want to apply the same criterias for all items
+            # so we merge the base domain
+            tag_ids = tag_ids or item.tag_ids.ids
+            if tag_ids:
+                tags_domain = [
+                    ('tag_ids', 'in', tag_ids),
+                ] + base_domain
+                # plus we exclude the current item
+                # if we are looking up by path
+                if path is None:
+                    tags_domain.append(('id', '!=', item.id, ))
+
+        # prepare final domain
+        # XXX: this could be done better...
+        domain = []
+        if tags_domain:
+            if len(base_domain + hierarchy_domain) > 1:
+                domain = ['|', '&', ] + \
+                    base_domain + hierarchy_domain
+            else:
+                domain = ['|', ] + \
+                    base_domain + hierarchy_domain
+            if len(tags_domain) > 1:
+                domain += ['&'] + tags_domain
+            else:
+                domain += tags_domain
+        else:
+            domain = base_domain + hierarchy_domain
+
+        order = order or 'sequence asc'
+        pages = self.search(domain, order=order)
+        return pages
+
+    def pager(self, total, page=1, step=10,
```

Comment:
I think we should simply drop this and use [the default pager](https://github.com/odoo/odoo/blob/9.0/addons/website/models/website.py#L423-L469) instead, which BTW is compatible with [`website_canonical_url`](https://github.com/OCA/website/tree/8.0/website_canonical_url) by default.